src/modules/sync/validation/elementSchemas.ts

```diff
@@ -0,0 +1,323 @@
+  is_non_localizable: z.boolean().optional(),
+}) satisfies z.ZodType<SyncSubpagesElement>;
+
+export const SnippetElementDataSchema = z.strictObject({
+  codename: z.string(),
+  type: z.literal("snippet"),
+  snippet: CodenameReferenceSchema,
+}) satisfies z.ZodType<SyncTypeSnippetElement>;
+
+export const TaxonomyElementDataSchema = z.strictObject({
+  name: z.string(),
+  codename: z.string(),
+  type: z.literal("taxonomy"),
+  taxonomy_group: CodenameReferenceSchema,
+  guidelines: z.string().optional(),
+  is_required: z.boolean().optional(),
+  is_non_localizable: z.boolean().optional(),
+  term_count_limit: z
+    .strictObject({
+      value: z.number(),
+      condition: z.union([z.literal("at_most"), z.literal("exactly"), z.literal("at_least")]),
+    })
+    .optional(),
+  default: z
+    .strictObject({
+      global: z.strictObject({
+        value: z.array(CodenameReferenceSchema),
+      }),
+    })
+    .optional(),
+}) satisfies z.ZodType<SyncTaxonomyElement>;
+
+export const TextElementDataSchema = z.strictObject({
+  name: z.string(),
+  codename: z.string(),
+  type: z.literal("text"),
+  is_required: z.boolean().optional(),
+  is_non_localizable: z.boolean().optional(),
+  guidelines: z.string().optional(),
+  maximum_text_length: z
+    .strictObject({ value: z.number(), applies_to: z.union([z.literal("words"), z.literal("characters")]) })
+    .optional(),
+  default: z.strictObject({ global: z.strictObject({ value: z.string() }) }).optional(),
+  validation_regex: z
+    .strictObject({
+      is_active: z.boolean(),
+      regex: z.string(),
+      flags: z.string().nullable().optional(),
+      validation_message: z.string().optional(),
+    }).optional(),
+}) satisfies z.ZodType<SyncTextElement>;
+
+export const UrlSlugElementDataSchema = z.strictObject({
+  name: z.string(),
+  codename: z.string(),
+  type: z.literal("url_slug"),
+  depends_on: z.strictObject({
+    element: CodenameReferenceSchema,
+    snippet: CodenameReferenceSchema.optional(),
+  }),
+  is_required: z.boolean().optional(),
+  is_non_localizable: z.boolean().optional(),
+  guidelines: z.string().optional(),
+  validation_regex: z
+    .strictObject({
+      is_active: z.boolean(),
+      regex: z.string(),
+      flags: z.string().optional(),
+      validation_message: z.string().optional(),
+    })
+    .optional(),
+}) satisfies z.ZodType<SyncUrlSlugElement>;
+
+export const SnippetElementsSchemas = [
+  AssetElementDataSchema,
+  CustomElementDataSchema,
+  DateTimeElementDataSchema,
+  GuidelinesElementDataSchema,
+  LinkedItemsElementDataSchema,
+  NumberElementDataSchema,
+  MultipleChoiceElementDataSchema,
+  RichTextElementDataSchema,
+  TaxonomyElementDataSchema,
+  TextElementDataSchema,
+] as const;
+
+const extendTuples = <Tuple1 extends ReadonlyArray<unknown>, Tuple2 extends ReadonlyArray<unknown>>(
+  t1: Tuple1,
+  t2: Tuple2,
+): CombineTuples<Tuple1, Tuple2> => [...t1, ...t2] as unknown as CombineTuples<Tuple1, Tuple2>;
```

Comment:
The casting is not necessary and in fact, the whole `extendTuples` is not needed. You can do the plain `[...SnippetElementsSchemas, SnipetElementDataSchema, ...] as const` and it works the same.